optimizely/helpers/condition.py

```diff
@@ -233,12 +238,60 @@ def substring_evaluator(self, index):
 
         return condition_value in user_value
 
+    def semver_equal_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) == 0
+
+    def semver_greater_than_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) == 1
+
+    def semver_less_than_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) == -1
+
+    def semver_less_than_and_equal_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) <= 0
+
+    def semver_greater_than_and_equal_evaluator(self, index):
+
+        return self.compare_user_version_with_target_version(index) >= 0
+
+    def compare_user_version_with_target_version(self, index):
+
+        condition_name = self.condition_data[index][0]
+        target_version = self.condition_data[index][1]
+        user_version = self.attributes.get(condition_name)
+
+        target_version_parts = target_version.split(".")
+        user_version_parts = user_version.split(".")
+
+        for (idx, _) in enumerate(target_version_parts):
+            if len(user_version_parts) <= idx:
+                return -1
+            # compare strings e.g: n1.n2.n3-alpha/beta
+            elif not str(user_version_parts[idx]).isdigit():
+                if user_version_parts[idx] != target_version_parts[idx]:
+                    return -1
+            # compare numbers e.g: n1.n2.n3
+            elif int(user_version_parts[idx]) > int(target_version_parts[idx]):
```

Comment:
Please check with Owais, `int` parsing twice time for the same thing, doesn't look OK. I know it's same as swift, but check if we can have better version.